### PR TITLE
CLN: Move base.StringMixin to computations.common

### PR DESCRIPTION
**Pull request by topper-123**
``StringMixin`` should be removed, but I can't figure out how to remove it without breaking tests, so this moves ``StringMixin`` to core.computation, which is the only module, where it is currently used.

This makes ``core.base.py`` a bit cleaner.

### CODE REVIEW

File: pandas/core/computation/common.py [hunk 1]
```diff
@@ -36,3 +36,8 @@ def _remove_spaces_column_name(name):
 
 class NameResolutionError(NameError):
     pass
+
+
+class StringMixin:
+    # TODO: delete this class. Removing this ATM caused a failure.
```
Comment (jbrockmendel) [inline pandas/core/computation/common.py]: what does it cause to fail?

Is there a reason you are using "..." instead of "pass"?

**Comment by jbrockmendel** [comment]
Generally +1.  If you were inclined to try to separate out the just-for-`io.pytables` parts I'd offer moral support for that too.

File: pandas/core/computation/common.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/computation/common.py]: Yeah, ``pass`` is more normal, I'll chnage that.

Comment (topper-123) [reply pandas/core/computation/common.py]: If you remove the inheriance from ``StringMixin`` in ``core.computatons.scope.py::Scope`` tests will fail. Haven't been able to fix it.

**Comment by topper-123** [comment]
I’ve updated the PR.

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**